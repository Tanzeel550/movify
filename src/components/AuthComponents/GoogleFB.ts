import React from 'react';
import { startLoginWithGoogle } from '../../actions/authActions';

const GoogleFB = () => (
  <div className="flex-c-m">
    <button className="social__item facebook-icon">
      <i className="zmdi zmdi-facebook" />
    </button>

    <button className="social__item google-icon" onClick={startLoginWithGoogle}>
      <i className="zmdi zmdi-google" />
    </button>
  </div>
);

export default GoogleFB;
