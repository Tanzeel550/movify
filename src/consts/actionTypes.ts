export type EmailPassType = {
  email: string;
  password: string;
};

export type AuthActionsType = {
  type: 'LOGIN' | 'LOGOUT';
  user?: object;
};

export type AuthUserType = {
  user: object;
};

export type MoviesActionReturnType = {
  type: 'CREATE_MOVIE' | 'GET_MOVIES' | 'UPDATE_MOVIE' | 'DELETE_MOVIE';
  movie?: object;
  movies?: object;
  id?: string;
};

export type MoviesActionParamsType = {
  id?: string;
  movie?: object;
  movies?: object;
};

export type SEARCH_TYPE = {
  text: string;
};

export type ErrorType = {
  message: string;
  title: string;
};

export type ErrorActionsReturnType = {
  type: 'SET_ERROR' | 'CLEAR_ERROR';
  message?: string;
  title?: string;
};
