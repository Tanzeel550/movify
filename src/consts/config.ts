const OMBD_API_KEY = process.env.REACT_APP_OMDB_API_KEY;
export const SEARCH_MOVIE_BY_TEXT = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&s=`;
export const GET_MOVIE_BY_TITLE = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&t=`;
export const GET_MOVIE_BY_IMDB_ID = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&i=`;

export const PROMISE_AREAS = {
  SEARCH_MOVIE_BY_TEXT: 'SEARCH_MOVIE_BY_TEXT',
  GET_MOVIE_BY_TITLE: 'GET_MOVIE_BY_TITLE',
  GET_ALL_MOVIES: 'GET_ALL_MOVIES',
  LOAD_PAGE: 'LOAD_PAGE',
};

export const skills1 = [
  'Web Design',
  'JavaScript Developer',
  'React JS',
  'Rest API',
  'NodeJS',
];

export const skills2 = ['Company', 'Team', 'Careers'];

export const TEST_MOVIES_SRC = 'movies_test';
