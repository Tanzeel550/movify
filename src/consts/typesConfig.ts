export type EmailPassType = {
  email: string;
  password: string;
};

export type AuthActionsType = {
  type: 'LOGIN' | 'LOGOUT';
  user?: object;
};

export type AuthUserType = {
  user: object;
};

export type MOVIES_ACTION = {
  type: 'CREATE_MOVIE' | 'GET_MOVIES' | 'UPDATE_MOVIE' | 'DELETE_MOVIE';
  movie?: object;
  movies?: object;
  id?: string;
};

export type SEARCH_TYPE = {
  text: string;
};

export type MOVIE_ID = {
  id?: string;
  movie?: object;
  movies?: object;
};
