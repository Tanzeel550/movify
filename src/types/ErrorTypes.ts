// ============================================================
// ======================= Error Types ========================
// ============================================================

export type ErrorState = {
  message: string;
  title: string;
};

export type ErrorActionsReturnType = {
  type: 'SET_ERROR' | 'CLEAR_ERROR';
  message?: string;
  title?: string;
};
