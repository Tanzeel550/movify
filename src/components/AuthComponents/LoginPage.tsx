import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import validator from 'validator';

import GoogleFB from './GoogleFB';
import { startLoginWithEmailPass } from '../../actions/authActions';
import { setError } from '../../actions/errorActions';
import { connect } from 'react-redux';

type Props = {
  setError: (...props: any) => {};
};

const LoginPage: React.FC<Props> = ({ setError }: Props) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleFormSubmit = async (e: React.FormEvent): Promise<void> => {
    e.preventDefault();
    try {
      if (!validator.isEmail(email))
        throw new Error(
          'This email is invalid. Please type a correct email...'
        );

      await startLoginWithEmailPass({ email, password });

      setEmail('');
      setPassword('');
    } catch (e) {
      setError({ message: e.message, title: 'Login' });
    }
  };

  const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setEmail(e.target.value);
  };

  const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setPassword(e.target.value);
  };

  return (
    <div className="form__container">
      <div className="form__wrapper">
        <form onSubmit={handleFormSubmit}>
          <span className="form__title text-center">Login</span>

          <div
            className="form__input--wrapper"
            data-validate="Username is required"
          >
            <span className="form__input--label">Username</span>
            <input
              className="form__input--input"
              type="email"
              name="username"
              placeholder="Type your username"
              value={email}
              onChange={handleEmailChange}
              required
            />
            <span className="form__input--icon" data-symbol="&#xf206;" />
          </div>

          <div
            className="form__input--wrapper "
            data-validate="Password is required"
          >
            <span className="form__input--label">Password</span>
            <input
              className="form__input--input"
              type="password"
              name="pass"
              placeholder="Type your password"
              value={password}
              onChange={handlePasswordChange}
              required
            />
            <span className="form__input--icon" data-symbol="&#xf190;" />
          </div>

          {/*<div className="text-right forgot-password">*/}
          {/*    <a href="#"> Forgot password? </a>*/}
          {/*</div>*/}

          <div className="form__button--container">
            <div className="form__button--wrapper">
              <div className="form__button--behind" />
              <button className="form__button--actual">Login</button>
            </div>
          </div>

          <div className="text-center txt1">
            <span> Or Login Using </span>
          </div>
          <GoogleFB />
          <div className="flex-col-c">
            <span>Or Sign Up Using</span>

            <NavLink to="/signUp" className="txt2">
              Sign Up
            </NavLink>
          </div>
        </form>
      </div>
    </div>
  );
};

export default connect(null, { setError })(LoginPage);
