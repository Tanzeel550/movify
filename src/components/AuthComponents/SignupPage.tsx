import React, { useState } from 'react';
import validator from 'validator';
import { NavLink } from 'react-router-dom';
import GoogleFB from './GoogleFB';
import { startSignUpWithEmailPass } from '../../actions/authActions';
import { connect } from 'react-redux';
import { setError } from '../../actions/errorActions';
import { ErrorActionsReturnType } from '../../consts/actionTypes';

type Props = {
  setError: ({
    message,
    title,
  }: {
    message: string;
    title: string;
  }) => ErrorActionsReturnType;
};

const SingUpPage: React.FC<Props> = ({ setError }: Props) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');

  const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setEmail(e.target.value);
  };

  const handlePasswordChange = (
    e: React.ChangeEvent<HTMLInputElement>
  ): void => {
    setPassword(e.target.value);
  };

  const handlePasswordConfirmChange = (
    e: React.ChangeEvent<HTMLInputElement>
  ): void => {
    setConfirmPassword(e.target.value);
  };

  const handleFormSubmit = async (e: React.FormEvent): Promise<void> => {
    e.preventDefault();
    try {
      if (!validator.isEmail(email))
        throw new Error(
          'This email is invalid. Please type a correct email...'
        );
      else if (password !== confirmPassword)
        throw new Error("Passwords don't match.");

      await startSignUpWithEmailPass({ email, password });
    } catch (e) {
      setError({ message: e.message, title: 'Signup' });
    }
  };

  return (
    <div className="form__container">
      <div className="form__wrapper">
        <form onSubmit={handleFormSubmit}>
          <span className="form__title text-center">Sign Up</span>

          <div
            className="form__input--wrapper"
            movie-validate="Username is required"
          >
            <span className="form__input--label">Username</span>
            <input
              className="form__input--input"
              type="email"
              name="email"
              placeholder="Type your username"
              value={email}
              onChange={handleEmailChange}
              required
            />
            <span className="form__input--icon" movie-symbol="&#xf206;" />
          </div>

          <div
            className="form__input--wrapper"
            movie-validate="Password is required"
          >
            <span className="form__input--label">Password</span>
            <input
              className="form__input--input"
              type="password"
              name="password"
              placeholder="Type your password"
              value={password}
              onChange={handlePasswordChange}
              required
            />
            <span className="form__input--icon" movie-symbol="&#xf190;" />
          </div>

          <div
            className="form__input--wrapper"
            movie-validate="Confirm Password is required"
          >
            <span className="form__input--label">Confirm Password</span>
            <input
              className="form__input--input"
              type="password"
              name="confirmPassword"
              placeholder="Type your password to Confirm"
              value={confirmPassword}
              onChange={handlePasswordConfirmChange}
              required
            />
            <span className="form__input--icon" movie-symbol="&#xf190;" />
          </div>

          <div className="form__button--container">
            <div className="form__button--wrapper">
              <div className="form__button--behind" />
              <button className="form__button--actual">Sign Up</button>
            </div>
          </div>

          <div className="text-center txt1">
            <span> Or Sign Up Using </span>
          </div>
          <GoogleFB />
          <div className="flex-col-c">
            <span className="txt1">Or Login Using</span>

            <NavLink to="/login" className="txt2">
              Login
            </NavLink>
          </div>
        </form>
      </div>
    </div>
  );
};

export default connect(null, { setError })(SingUpPage);
