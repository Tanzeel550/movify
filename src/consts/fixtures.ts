import {
  APIMovieResultByTitle,
  APISingleMovieResultBySearch,
  FireDBMovieItem,
} from '../types/APITypes';
import { AuthUserType } from '../types/AuthTypes';

export const movies: FireDBMovieItem[] = [
  {
    id: '1234567',
    name: 'The Dark Knight',
    watched: true,
    dateWatched: '234567890',
    whatYouLearnt: 'The world is as good as we let it to be',
    Poster: 'https://image.com',
    createdAt: '9876543',
  },
  {
    id: '4567890',
    name: 'GodFather',
    watched: false,
    dateWatched: '98765434567',
    whatYouLearnt: 'You can act like a man what is the matter with you?',
    Poster: 'https://godfather.com',
    createdAt: '987654567890',
  },
  {
    id: '9ikjnbvfrty',
    name: 'Avengers: EndGame',
    watched: false,
    dateWatched: '',
    whatYouLearnt: 'Thanos will be defeated',
    Poster: 'https://avengers.com',
    createdAt: '69864123340404',
  },
];

export const myUser: AuthUserType = {
  user: {
    name: 'Tanzeel',
    email: 'tanzeelahmed550@gmail.com',
    passion: 'Web Dev',
    os: 'Kali Linux',
    version: '1.0.0',
    height: 5.5,
  },
};

export const skills1 = [
  'Web Design',
  'JavaScript Developer',
  'React JS',
  'Rest API',
  'NodeJS',
];

export const skills2 = ['Company', 'Team', 'Careers'];

export const email = 'unknownPerson123@uet.com';
export const password = 'JonSnow';
export const nameOfTheMovie = 'Godfather';

export const movieFromAPIByTitle: APIMovieResultByTitle = {
  Title: 'The Dark Knight',
  Year: '2008',
  Rated: 'PG-13',
  Released: '18 Jul 2008',
  Runtime: '152 min',
  Genre: 'Action, Crime, Drama, Thriller',
  Director: 'Christopher Nolan',
  Writer:
    'Jonathan Nolan (screenplay), Christopher Nolan (screenplay), Christopher Nolan (story), David S. Goyer (story), Bob Kane (characters)',
  Actors: 'Christian Bale, Heath Ledger, Aaron Eckhart, Michael Caine',
  Plot:
    'When the menace known as the Joker wreaks havoc and chaos on the people of Gotham, Batman must accept one of the greatest psychological and physical tests of his ability to fight injustice.',
  Language: 'English, Mandarin',
  Country: 'USA, UK',
  Awards: 'Won 2 Oscars. Another 157 wins & 163 nominations.',
  Poster:
    'https://m.media-amazon.com/images/M/MV5BMTMxNTMwODM0NF5BMl5BanBnXkFtZTcwODAyMTk2Mw@@._V1_SX300.jpg',
  Ratings: [
    { Source: 'Internet Movie Database', Value: '9.0/10' },
    {
      Source: 'Rotten Tomatoes',
      Value: '94%',
    },
    { Source: 'Metacritic', Value: '84/100' },
  ],
  Metascore: '84',
  imdbRating: '9.0',
  imdbVotes: '2,341,244',
  imdbID: 'tt0468569',
  Type: 'movie',
  DVD: '14 Jun 2010',
  BoxOffice: '$534,858,444',
  Production: 'Syncopy',
  Website: 'N/A',
  Response: 'True',
};

export const moviesFromAPIBySearch: APISingleMovieResultBySearch[] = [
  {
    Title: 'The Dark Knight',
    Year: '2008',
    imdbID: 'tt0468569',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMTMxNTMwODM0NF5BMl5BanBnXkFtZTcwODAyMTk2Mw@@._V1_SX300.jpg',
  },
  {
    Title: 'The Dark Knight Rises',
    Year: '2012',
    imdbID: 'tt1345836',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMTk4ODQzNDY3Ml5BMl5BanBnXkFtZTcwODA0NTM4Nw@@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Dark Knight Returns, Part 1',
    Year: '2012',
    imdbID: 'tt2313197',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMzIxMDkxNDM2M15BMl5BanBnXkFtZTcwMDA5ODY1OQ@@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Dark Knight Returns, Part 2',
    Year: '2013',
    imdbID: 'tt2166834',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BYTEzMmE0ZDYtYWNmYi00ZWM4LWJjOTUtYTE0ZmQyYWM3ZjA0XkEyXkFqcGdeQXVyNTA4NzY1MzY@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Dark Knight Returns',
    Year: '2013',
    imdbID: 'tt11060882',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMjExMGE4NDQtM2MwMS00YzhlLTlkYjgtNmExNzM1ZDBhYzg1XkEyXkFqcGdeQXVyMTY5Nzc4MDY@._V1_SX300.jpg',
  },
];
