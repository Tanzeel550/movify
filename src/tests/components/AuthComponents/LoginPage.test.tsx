import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { LoginPage } from '../../../components/AuthComponents/LoginPage';
import { ActionCreatorWithPayload } from '@reduxjs/toolkit';
import { ErrorState } from '../../../types/ErrorTypes';
import { email, password } from '../../../consts/fixtures';

// jest.setTimeout(120000); // 2 minutes = 12000 milliseconds

let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>,
  setError: ActionCreatorWithPayload<ErrorState, string> | jest.Mock<any, any>,
  startLoginWithEmailPass: jest.Mock<any, any>;

beforeEach(() => {
  setError = jest.fn();
  startLoginWithEmailPass = jest.fn();
  wrapper = shallow(
    <LoginPage
      // @ts-ignore
      setError={setError}
      startLoginWithEmailPass={startLoginWithEmailPass}
    />
  );
});

test('should render GoogleFB component correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should set email when emailText change', () => {
  const elem = wrapper.find('input').at(0);
  elem.simulate('change', { target: { value: email } });
  expect(wrapper.state('email')).toBe(email);
});

test('should set password when passwordText change', () => {
  wrapper
    .find('input')
    .at(1)
    .simulate('change', { target: { value: password } });
  expect(wrapper.state('password')).toBe(password);
});

test('should submit the login form successfully because email and password are valid', () => {
  wrapper.setState({ email, password });
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {},
  });
  expect(startLoginWithEmailPass).toBeCalledWith({ email, password });
  // expect(wrapper.state('email')).toBe('');
  // expect(wrapper.state('password')).toBe('');
});

test('should not submit the login form because email is invalid', () => {
  wrapper.setState({ email: 'unknownPerson123' });
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {},
  });

  expect(setError).toBeCalledWith({
    message: 'This email is invalid. Please type a valid email...',
    title: 'Login',
  });
});

test('should not submit the login form because password is invalid', () => {
  wrapper.setState({ email, password: '' });
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {},
  });
  expect(setError).toBeCalledWith({
    message: 'Please type a valid password',
    title: 'Login',
  });
});

export {};
