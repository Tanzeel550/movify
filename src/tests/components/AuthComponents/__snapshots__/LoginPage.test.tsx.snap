// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GoogleFB component correctly 1`] = `
<div
  className="form__container"
>
  <div
    className="form__wrapper"
  >
    <form
      onSubmit={[Function]}
    >
      <span
        className="form__title text-center"
      >
        Login
      </span>
      <div
        className="form__input--wrapper"
        movie-validate="Username is required"
      >
        <span
          className="form__input--label"
        >
          Username
        </span>
        <input
          className="form__input--input"
          name="username"
          onChange={[Function]}
          placeholder="Type your username"
          required={true}
          type="email"
          value=""
        />
        <span
          className="form__input--icon"
          movie-symbol=""
        />
      </div>
      <div
        className="form__input--wrapper "
        movie-validate="Password is required"
      >
        <span
          className="form__input--label"
        >
          Password
        </span>
        <input
          className="form__input--input"
          name="pass"
          onChange={[Function]}
          placeholder="Type your password"
          required={true}
          type="password"
          value=""
        />
        <span
          className="form__input--icon"
          movie-symbol=""
        />
      </div>
      <div
        className="form__button--container"
      >
        <div
          className="form__button--wrapper"
        >
          <div
            className="form__button--behind"
          />
          <button
            className="form__button--actual"
          >
            Login
          </button>
        </div>
      </div>
      <div
        className="text-center txt1"
      >
        <span>
           Or Login Using 
        </span>
      </div>
      <GoogleFB />
      <div
        className="flex-col-c"
      >
        <span>
          Or Sign Up Using
        </span>
        <NavLink
          className="txt2"
          to="/signUp"
        >
          Sign Up
        </NavLink>
      </div>
    </form>
  </div>
</div>
`;
